src/tokens/ERC721.sol

```diff
@@ -273,16 +282,13 @@ abstract contract ERC721 {
     //////////////////////////////////////////////////////////////*/
     
     function _mint(
-        address to, 
-        uint256 tokenId
-    ) internal virtual { 
-        require(ownerOf[tokenId] == address(0), "ALREADY_MINTED");
-  
-        // Cannot realistically overflow from incrementing total supply beyond
+        address to
+    ) internal virtual returns (uint256 tokenId) { 
+        // Cannot realistically overflow from minting beyond
         // the max uint256 value, and because the sum of all user balances 
         // can't exceed the max uint256 value.
         unchecked {
-            totalSupply++;
+            tokenId = minted++;
```

Comment:
Personally think should be idiomatic to start at 1 rather than 0 🙂
```suggestion
            tokenId = ++minted;
```

Reply:
TIL about that syntax woah